static/merge4.js

```diff
@@ -366,4 +371,181 @@
 
         remakeTooltips();
     })
+
+    $('.mergeVertical-4').click(function() {
+        const construct = $(this).data('construct');
+        const constructId = construct.split('-')[1];
+        const sublevels = $(this).data('sublevels');
+
+        // Columns with courses names in table.
+        const classesColumn = 1;
+
+        // Create object which let restore removed cells later.
+        createObjectToRestore(constructId, sublevels, classesColumn);
+
+        // Initialize array with 0 for every sublevel.
+        var observationQuantity = initQuantityArray(sublevels);
+        var sublevelsIds = [];
+
+        // Calculate amount of observations per sublevel and append their ids to `sublevelIds` array.
+        for (var i in window.verticalStarChart[constructId]) {
+            for (var j in window.verticalStarChart[constructId][i].quantities) {
+                if (j != "0") {
+                    observationQuantity[j - 1] += parseInt(
+                        window.verticalStarChart[constructId][i].quantities[j].dataset.stars);
+                    sublevelsIds.push($(window.verticalStarChart[constructId][i].classes[j])
```

Comment:
Try to avoid long expressions passed as function parameters. You can assign `window.vertical ...start` and `window.vertical ...data('sublevel')` to variables - it still should takes 4 lines.